stlab/concurrency/utility.hpp

```diff
@@ -182,6 +182,42 @@ inline void blocking_get(future<void> x) {
     if (error) std::rethrow_exception(error);
 }
 
+
+inline bool blocking_get(future<void> x, const std::chrono::nanoseconds& timeout) {
+    std::exception_ptr error = nullptr;
+
+    bool set{false};
+    std::condition_variable condition;
+    std::mutex m;
+    auto hold = std::move(x).recover(immediate_executor, [&](auto&& r) {
+        if (r.error()) error = std::forward<decltype(r)>(r).error().value();
+        {
+            std::unique_lock<std::mutex> lock(m);
```

Comment:
I think, all local variables of blocking_get() here and the other version have to be put into a shared struct and be passed with a weak_ptr into the recover function. If the timeout elapses and the blocking_get() leaves the scope and at that timepoint the recover function is run, then the access to the mutex is bad ;-)
Sorry I did not see this earlier.